src/datajudge/constraints/uniques.py

```diff
@@ -292,3 +294,76 @@ def compare(
     def test(self, engine: sa.engine.Engine) -> TestResult:
         self.max_relative_gain = self.max_relative_gain_getter(engine)
         return super().test(engine)
+
+
+class VariantDistributionConstraint(Constraint):
+    """
+    VariantDistributionConstraint is a class to check if the distribution of values in a column
+    falls within the specified minimum and maximum bounds.
+    The constraint compares the factual distribution of values in a column of a `DataSource`
+    with the target distribution supplied as a dictionary where keys represent the unique values
+    and the corresponding tuple values represent the minimum and maximum allowed shares of
+    the respective unique value in the column.
+
+    The constraint is initialized with a `DataReference` object, the target distribution,
+    a minimum and maximum bound for each value not in the given distribution and
+    optionally a name and any additional keyword arguments.
+
+    Example use cases include testing for consistency in columns with expected categorical values
+    or ensuring that the distribution of values in a column adheres to a certain criterion.
+    """
+
+    def __init__(
+        self,
+        ref: DataReference,
+        distribution: Dict[T, Tuple[float, float]],
+        default_bounds: Tuple[float, float] = (0, 0),
```

Comment:
What would you think of a relative violation tolerance parameter? E.g. it could say:

A test succeeds iff
#observations outside of the specified ranges / #observations <= tolerance_parameter

I don't consider it a must - we've simply faren well with tolerances historically.

Reply:
If 'A' is expected to have a target share ranging from 5% to 15%, but its actual share is 16%, would you consider the 16% to be a violation of the target range or merely 1% above the upper limit?

Reply:
I added the feature